### PR TITLE
Create basic tools for release

### PR DESCRIPTION
**Pull request by Czaki**

### CODE REVIEW

File: filter_opened_bug_issues.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+else:
+    milestone_search_string = ""
+    milestone = None
+
+
+common_ancestor = get_common_ancestor(args.from_commit, args.to_commit)
+remote_commit = repository.get_commit(common_ancestor.hexsha)
+previous_tag_date = datetime.strptime(
+    remote_commit.last_modified, "%a, %d %b %Y %H:%M:%S %Z"
+)
```
Comment (melonora) [inline filter_opened_bug_issues.py:58]: Isn't this the same as `get_split_date`?

File: filter_pr_that_may_be_selected.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        ]
+    else:
+        triage_labels = []
+
+    previous_tag_date = get_split_date(from_commit, to_commit)
+
+    if milestone is not None:
+        query = f"milestone:{milestone.title} is:merged "
+    else:
+        query = f"merged:>{previous_tag_date.isoformat()} no:milestone "
```
Comment (melonora) [inline filter_pr_that_may_be_selected.py:94]:
```suggestion
    if milestone is not None:
        query = f"milestone:{milestone.title} is:merged "
    else:
        previous_tag_date = get_split_date(from_commit, to_commit)
        query = f"merged:>{previous_tag_date.isoformat()} no:milestone "
```

File: find_contributors_without_citation.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    iter_pull_request,
+    setup_cache,
+)
+
+
+def existing_file(path: str) -> Path:
+    path = Path(path)
+    if not path.exists():
+        raise FileNotFoundError(f"{path} not found")
+    return path
```
Comment (melonora) [inline find_contributors_without_citation.py:34]:
```suggestion
```
Duplicated function -> move to `release_utils`.

File: find_contributors_without_citation.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+import argparse
+from pathlib import Path
+
+from tqdm import tqdm
+from yaml import safe_load
+
+from release_utils import (
+    BOT_LIST,
+    LOCAL_DIR,
+    REPO_DIR_NAME,
```
Comment (melonora) [inline find_contributors_without_citation.py:22]:
```suggestion
    REPO_DIR_NAME,
    existing_file,
```

File: add_login_to_citation_cff.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+LOCAL_DIR = Path(__file__).parent
+DEFAULT_CORRECTION_FILE = LOCAL_DIR / "name_corrections.yaml"
+
+
+def existing_file(value):
+    file_path = Path(value)
+    if not file_path.exists():
+        raise argparse.ArgumentTypeError(f"The file {file_path} does not exists.")
+    return file_path
```
Comment (melonora) [inline add_login_to_citation_cff.py:26]:
```suggestion
```
Duplicate code -> move to `release_utils`.

File: add_login_to_citation_cff.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+import argparse
+import shutil
+import sys
+from pathlib import Path
+
+from tqdm import tqdm
+from unidecode import unidecode
+from yaml import safe_dump, safe_load
+
+from release_utils import BOT_LIST, get_repo, setup_cache
```
Comment (melonora) [inline add_login_to_citation_cff.py:16]:
```suggestion
from release_utils import BOT_LIST, existing_file, get_repo, setup_cache
```

File: generate_release_notes.py [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+
+
+setup_cache()
+repo = get_repo()
+
+correction_dict = {}
+with open(args.correction_file) as f:
+    corrections = safe_load(f)
+    for correction in corrections["login_to_name"]:
+        correction_dict[correction["login"]] = correction["corrected_name"]
```
Comment (melonora) [inline generate_release_notes.py:67]: This seems very similar to `get_correction_dict` in `add_login_to_citation_cff.py`.

**Review by melonora — COMMENTED** [review]
I couldn't yet try out the different tools, but I did go through the code. Thanks for creating these tools! Overall, I can follow the logic, but I think that as we work with the tools we can add some more documentation in order to reduce the time that somebody requires to get familiar with it.

**Comment by Czaki** [comment]
Thanks for review and spot bugs that I made when try to refactor cod in bunch of functions to increase readability.

File: add_login_to_citation_cff.py [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+        correction_file
+    ) | get_corrections_from_citation_cff(data)
+
+    contributors = {
+        get_name(user, correction_dict): user
+        for user in tqdm(contributors_iterable, total=contributors_iterable.totalCount)
+        if get_name(user, correction_dict) is not None
+    }
+
+    for user in get_repo().get_contributors():
```
[the remark this answers is not in the document]
Comment (Czaki) [reply add_login_to_citation_cff.py:69]: maybe I'm wrong. But I'm pretty sure that I meet this as a problem. I will check it when work again with this script (as now all logins are added).

**Pull request merged**